test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -2251,8 +2251,15 @@ class ClientLbInterceptTrailingMetadataTest : public ClientLbEnd2endTest {
     return std::move(load_report_);
   }
 
-  grpc::internal::Mutex mu_;
-  grpc::internal::CondVar cond_;
+  bool WaitForLbCallback() {
+    grpc::internal::MutexLock lock(&mu_);
+    bool timed_out = false;
```

Comment:
I don't think we need this bool.  If the wait times out, then we know that there wasn't a callback, so there's no need to set `trailer_intercepted_` to false, which means that we can return immediately.

```
while (!trailer_intercepted_) {
  if (!cond_.WaitWithTimeout(&mu_, absl::Seconds(3))) return false;
}
trailer_intercepted_ = false;
return true;
```

Reply:
done